crates/sui-core/src/transaction_driver/mod.rs

```diff
@@ -105,6 +108,8 @@ where
                             TX_TYPE_SHARED_OBJ_TX
                         }])
                         .observe(settlement_finality_latency);
+                    // Record the number of retries
+                    self.metrics.transaction_retries.observe(attempts as f64);
```

Comment:
What do you think about adding a flag for result (success vs failure) and then track # of retries before getting to that final result?